com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/resource/CheckBatchLinkableResourceStorageFacade.java

```diff
@@ -26,4 +60,31 @@ public void saveResource(final StorageAwareResource resource, final IFileSystemA
       throw e;
     }
   }
+
+  @Override
+  protected boolean doesStorageExist(final StorageAwareResource resource) {
+    URI uri = getBinaryStorageURI(resource);
+    ResourceSet resourceSet = resource.getResourceSet();
+
+    return (uri != null && resourceSet.getURIConverter().exists(uri, null)) || super.doesStorageExist(resource);
+  }
+
+  @Override
+  public ResourceStorageLoadable getOrCreateResourceStorageLoadable(final StorageAwareResource resource) {
+    try {
+      URI storageURI = getBinaryStorageURI(resource);
+      if (storageURI == null) {
+        return super.getOrCreateResourceStorageLoadable(resource);
+      }
+
+      URIConverter uriConverter = resource.getResourceSet().getURIConverter();

```

Comment:
getResourceSet() could we null, we need to check for that and call super.... then